src/abstractitensornetwork.jl

```diff
@@ -585,17 +586,22 @@ function LinearAlgebra.factorize(tn::AbstractITensorNetwork, edge::Pair; kwargs.
 end
 
 # For ambiguity error; TODO: decide whether to use graph mutating methods when resulting graph is unchanged?
-function orthogonalize_walk(tn::AbstractITensorNetwork, edge::AbstractEdge; kwargs...)
-  return orthogonalize_walk(tn, [edge]; kwargs...)
+function gauge_walk(
+  alg::Algorithm, tn::AbstractITensorNetwork, edge::AbstractEdge; kwargs...
+)
+  return gauge_walk(tn, [edge]; kwargs...)
 end
 
-function orthogonalize_walk(tn::AbstractITensorNetwork, edge::Pair; kwargs...)
-  return orthogonalize_walk(tn, edgetype(tn)(edge); kwargs...)
+function gauge_walk(alg::Algorithm, tn::AbstractITensorNetwork, edge::Pair; kwargs...)
+  return gauge_walk(alg::Algorithm, tn, edgetype(tn)(edge); kwargs...)
 end
 
 # For ambiguity error; TODO: decide whether to use graph mutating methods when resulting graph is unchanged?
-function orthogonalize_walk(
-  tn::AbstractITensorNetwork, edges::Vector{<:AbstractEdge}; kwargs...
+function gauge_walk(
```

Comment:
Can we make this function generic to any gauge, and have it loop over a lower level function `gauge_edge(alg::Algorithm, tn::AbstractITensorNetwork, edge::AbstractEdge)` that is then specialized to the gauge? It seems like most gauges will reduce to looping over edges.

Reply:
Makes sense, I changed it to that!